src/Sylius/Bundle/CartBundle/Provider/CartProvider.php

```diff
@@ -130,6 +138,15 @@ private function provideCart()
         }
 
         $cart = $this->cartFactory->createNew();
+
+        $event = new GenericEvent($cart);
+        $this->eventDispatcher->dispatch(SyliusCartEvents::CART_INITIALIZE, $event);
+
+        $cart = $event->getSubject();
+
+        $this->entityManager->persist($cart);
+        $this->entityManager->flush();
```

Comment:
We talked about using a separate sequence but that seemed like overkill. The way this all works kinda needs this id to exist, open to other ideas here, but unless there's a major upheaval of structure you really need to flush.